integration-tests/src/test/java/org/apache/fineract/integrationtests/DateValidationTest.java

```diff
@@ -0,0 +1,237 @@
+                .build(clientId.toString(), loanProductID.toString(), null);
+        HashMap<String, Object> response = (HashMap) loanTransactionHelper.createLoanAccount(loanApplicationJSON, "");
+        List<HashMap<String, Object>> errors = (List) response.get("errors");
+        assertNotNull(errors);
+        HashMap<String, Object> error = errors.get(0);
+        assertNotNull(error);
+        assertEquals(
+                "The parameter `expectedDisbursementDate` is invalid based on the dateFormat: `dd MMMM uuuu` and locale: `en_GB` provided:",
+                error.get("developerMessage"));
+    }
+
+    @Test
+    public void testShouldFailWithInvalidDateTime() {
+        String requestCode = UUID.randomUUID().toString();
+        InteropTransactionRole role = InteropTransactionRole.PAYER;
+        String requestBody = buildRequestBody(requestCode, role);
+        String response = interopHelper.postTransactionRequest(requestCode, role, requestBody);
+        HashMap<String, Object> map = new Gson().fromJson(response, new TypeToken<HashMap<String, Object>>() {}.getType());
+        List<Map<String, Object>> errors = (List) map.get("errors");
+        assertNotNull(errors);
+        Map<String, Object> error = errors.get(0);
+        assertNotNull(error);
+        assertEquals("The parameter `expiration` is invalid based on the dateFormat: `dd MMMM uuuu HH:mm:ss` and locale: `en` provided:",
+                error.get("developerMessage"));
+    }
+
+    @Test
+    public void testShouldFailWithInvalidMonthDay() {
+        final Account assetAccount = this.accountHelper.createAssetAccount();
+        final Account incomeAccount = this.accountHelper.createIncomeAccount();
+        final Account expenseAccount = this.accountHelper.createExpenseAccount();
+        final Account liabilityAccount = this.accountHelper.createLiabilityAccount();
+
+        DateFormat dateFormat = new SimpleDateFormat("dd MMMM yyyy", Locale.US);
+        new SimpleDateFormat("dd MMM", Locale.US);
+        DateFormat currentDateFormat = new SimpleDateFormat("dd");
+
+        Calendar todaysDate = Calendar.getInstance();
+        todaysDate.add(Calendar.MONTH, -3);
+        final String VALID_FROM = dateFormat.format(todaysDate.getTime());
+        todaysDate.add(Calendar.YEAR, 10);
+        final String VALID_TO = dateFormat.format(todaysDate.getTime());
+
+        todaysDate = Calendar.getInstance();
+        todaysDate.add(Calendar.MONTH, -1);
+        final String SUBMITTED_ON_DATE = dateFormat.format(todaysDate.getTime());
+
+        Integer currentDate = Integer.valueOf(currentDateFormat.format(todaysDate.getTime()));
+        Integer daysInMonth = todaysDate.getActualMaximum(Calendar.DATE);
+        Integer numberOfDaysLeft = daysInMonth - currentDate + 1;
+        todaysDate.add(Calendar.DATE, numberOfDaysLeft);
+
+        PostClientsRequest postClientsRequest = ClientHelper.defaultClientCreationRequest();
```

Comment:
Please remove all the parts that are not relevant for the test case!

Reply:
I need a client ID for the test request

Reply:
that was more of a general request, not particularly for this line ;)